siriuspy/siriuspy/idff/main.py

```diff
@@ -0,0 +1,226 @@
+"""ID feedforward App."""
+
+import os as _os
+import logging as _log
+import time as _time
+import epics as _epics
+
+from ..callbacks import Callback as _Callback
+from ..devices import IDFF as _IDFF
+
+from .csdev import IDFFConst as _Const, ETypes as _ETypes
+
+
+class App(_Callback):
+    """Main application for handling machine shift."""
+
+    def __init__(self, idname):
+        """Class constructor."""
+        super().__init__()
+        self._const = _Const(idname)
+        self._pvs_prefix = self._const.idffname
+        self._pvs_database = self._const.get_propty_database()
+
+        self._loop_state = _Const.DsblEnbl.Dsbl
+        self._loop_freq = _Const.DEFAULT_LOOP_FREQ
+        self._control_qs = _Const.DsblEnbl.Dsbl
+        self._polarization = 'none'
+        self._config_name = ''
+        self.read_autosave_file()
+
+        # connections
+        # use pyepics recommendations for threading
+        _epics.ca.use_initial_context()
+
+        self._idff = _IDFF(idname)
+
+        # pvs to write methods
+        self.map_pv2write = {
+            'LoopState-Sel': self.set_loop_state,
+            'LoopFreq-SP': self.set_loop_freq,
+            'ControlQS-Sel': self.set_control_qs,
+            'ConfigName-SP': self.set_config_name,
+        }
+
+        self.quit = False
+        self._thread_ff = _epics.ca.CAThread(
+            target=self._do_ff, daemon=True)
+        self._thread_ff.start()
+
+    def init_database(self):
+        """Set initial PV values."""
+        pvn2vals = {
+            'LoopState-Sel': self._loop_state,
+            'LoopState-Sts': self._loop_state,
+            'LoopFreq-SP': self._loop_freq,
+            'LoopFreq-RB': self._loop_freq,
+            'ControlQS-Sel': self._control_qs,
+            'ControlQS-Sts': self._control_qs,
+            'ConfigName-SP': self._config_name,
+            'ConfigName-RB': self._config_name,
+            'Polarization-Mon': self._polarization,
+        }
+        for pvn, val in pvn2vals.items():
+            self.run_callbacks(pvn, val)
+        self._load_config(self._config_name)
```

Comment:
@anacso17 , why did you include here the loading of the IDFF? as it is now it will always fail for config name 'none' in the IOC initialization, right? do you intend to change the default config name  from 'none' to another configuration in the future? if it is the case I suggest either moving this loading to a separate initialization section of renaming the method from `init_database` to something more general, `init`.

Reply:
I see, @anacso17 , you are updating _config_name within `read_autosave_file` invoked previously. Then I suggest renaming `init_database` or loaginf the IDFF config in a separate section/method.

Reply:
done!